articles/introduction-to-c#/index .md

```diff
@@ -0,0 +1,212 @@
+
+**Line 3:**
+ ``` C#
+ class Program{}
+ ```
+
+A class is a container that contains methods and data. Since C# is Object-oriented programming, creating a class is mandatory for each program. In our case, we are creating a class called `Program`.
+
+**Line 4:** 
+```C#
+static void Main(string[] args)
+```
+Execution of a program begins from the `Main` method. Every C# program must have the `Main` method.
+
+**Line 5:**
+ ```C#
+ Console.WriteLine("Hello World!");
+ ```
+
+`Console.WriteLine()` is used to print out text on the console. It prints a string and moves to the next line. In our case, it is used to output *"Hello World"*.
+
+We can also use `Console.Write()` to print a string in the same line without moving to the next line. You can learn more on `Console.WriteLine()` and `Console.Write()` [here](https://www.programiz.com/csharp-programming/basic-input-output).
+
+### Reading character(s) from the input:
+`Console.ReadLine()` is used to read the next characters in the input system.
+
+`Console. Read()` is used to read the next character in the input system.
+
+`Console.ReadKey()` is used to obtain the next key entered by the user. It is mostly used to hold the console for the user until he presses a key.
+
+### Comments in C#
+Comments start with two forward slashes. The compiler ignores comments when compiling the code.
+
+**Example**
+```c#
+// This is a C# program
+```
+
+This is a single-line comment. Comments are used by programmers to explain more about the line (what the line does). This concept is very important when a group of programmers is working together as it makes your code more readable to the others.
+
+For multiple-line comment, we use ` /*` and ends with `*/` :
+
+```C#
+/* C# is a programming language used to create Web applications and Mobile applications.
+
+It is an amazing language. */
+```
+
+Any text between `/*` and `*/` will be ignored by the compiler.
+
+**Note:**
+- Each line in C# ends with a semi-colon(;).
+- C# follows a top to bottom order when executing the program.
+- C# is case-sensitive as it treats Uppercase and Lowercase characters differently. e.g `program` and `Program` are different things.
+
+**Congratulations!** You just wrote your first C# program.
+
+Now, let's write a program that adds two numbers entered by the user.
+
+```C#
+//This is a C# program that adds two numbers
+
+using System;
+
+    namespace HelloWorld
+        {
+            class Program
+            {       
+                static void Main(string[] args){
+                    Console.WriteLine("Enter your first number: ");
+                    int num1= Convert.ToInt32(Console.ReadLine());
+                    Console.WriteLine("Enter your second number: ");
+                    int num2= Convert.ToInt32(Console.ReadLine());
+                    Console.WriteLine("Your Sum is "num1 + num2);//prints the sum of the two numbers 
+                    Console.ReadLine();
+               }
+         }
+    }
+```
```

Comment:
This code doesn't work as intended. 
When I enter 5 and 7, the output is 57.

What's happening is, when you use a `+` operator on a string, it means it concatenates. The code is concatenating `"Your sum is "`, `num1`,  and `num2`.

You should either wrap `num1` and `num2` in `()` like this `Console.WriteLine("Your Sum is " + (num1 + num2))`
or introduce a new variable for the sum, and use it to print the answer.

Like this,
```C#
int sum = num1 + num2;
Console.WriteLine("Your Sum is " + sum)
```

This option is much better since it is very clear what is happening.